gammapy/spectrum/spectrum_dataset.py

```diff
@@ -0,0 +1,205 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from pathlib import Path
+import numpy as np
+from astropy.units import Quantity
+from ..utils.scripts import make_path
+from ..irf import EffectiveAreaTable, EnergyDispersion
+from .core import PHACountsSpectrum
+from .utils import SpectrumEvaluator
+from ..utils.fitting import Dataset, Parameters
+from..stats import wstat
+
+__all__ = [
+    "ONOFFSpectrumDataset"
+]
+
+
+class ONOFFSpectrumDataset(Dataset):
+    """Compute spectral model fit statistic on a ON OFF Spectrum.
+
+
+    Parameters
+    ----------
+    model : `~gammapy.spectrum.models.SpectralModel`
+        Fit model
+    ONcounts : `~gammapy.spectrum.PHACountsSpectrum`
```

Comment:
Again I think we rather use suffixes...so maybe rename to `counts_on` and `counts_off`?

Reply:
OK